python_modules/dagster/dagster/_core/definitions/sensor_definition.py

```diff
@@ -154,6 +159,120 @@ def repository_name(self) -> Optional[str]:
+        last_run_key (str): DEPRECATED The run key of the RunRequest most recently created by this
+            sensor. Use the preferred `cursor` attribute instead.
+        repository_name (Optional[str]): The name of the repository that the sensor belongs to.
+        instance (Optional[DagsterInstance]): The deserialized instance can also be passed in
+            directly (primarily useful in testing contexts).
+
+    Example:
+
+    .. code-block:: python
+
+        from dagster import multi_asset_sensor, MultiAssetSensorEvaluationContext
+
+        @multi_asset_sensor(asset_keys=[AssetKey("asset_1), AssetKey("asset_2)])
+        def the_sensor(context: MultiAssetSensorEvaluationContext):
+            ...
+
+    """
+
+    def __init__(
+        self,
+        instance_ref: Optional[InstanceRef],
+        last_completion_time: Optional[float],
+        last_run_key: Optional[str],
+        cursor: Optional[Mapping[AssetKey, str]],
+        repository_name: Optional[str],
+        asset_keys: Sequence[AssetKey],
+        instance: Optional[DagsterInstance] = None,
+    ):
+        self._asset_keys = asset_keys
+        self.cursor_has_been_updated = False
+
+        super(MultiAssetSensorEvaluationContext, self).__init__(None, None, None, None, None, None)
+
+        self._exit_stack = ExitStack()
+        self._instance_ref = check.opt_inst_param(instance_ref, "instance_ref", InstanceRef)
+        self._last_completion_time = check.opt_float_param(
+            last_completion_time, "last_completion_time"
+        )
+        self._last_run_key = check.opt_str_param(last_run_key, "last_run_key")
+        self._cursor = check.opt_dict_param(cursor, "cursor", key_type=AssetKey, value_type=str)
+        self._repository_name = check.opt_str_param(repository_name, "repository_name")
+        self._instance = check.opt_inst_param(instance, "instance", DagsterInstance)
+
+    def latest_materializations_by_key(self, asset_keys: Optional[Sequence[AssetKey]] = None):
+        from dagster._core.events import DagsterEventType
+        from dagster._core.storage.event_log.base import EventRecordsFilter
+
+        if asset_keys is None:
+            asset_keys = self._asset_keys
+
+        cursor_dict = json.loads(self.cursor) if self.cursor else {}
+        asset_event_records = {}
+        for a in asset_keys:
+            event_records = self.instance.get_event_records(
+                EventRecordsFilter(
+                    event_type=DagsterEventType.ASSET_MATERIALIZATION,
+                    asset_key=a,
+                    after_cursor=cursor_dict.get(a),
+                ),
+                ascending=False,
+                limit=1,
+            )
+
+            if event_records:
+                asset_event_records[a] = event_records[0]
+            else:
+                asset_event_records[a] = None
+
+        return asset_event_records
+
+    def materializations_for_key(self, asset_key: AssetKey, limit: int):
```

Comment:
Doesn't need to be this diff, but we potentially will want to expose `ascending` as an argument.  This might affect the `advance_cursor` API a little so at least worth thinking through a couple cases.

Reply:
I don't think exposing `ascending` would cause any problems because the user has to provide a single EventLogRecord to `advance_cursor`, so they will have to make a dictionary where they pick the record they want the cursor to advance to. I adding a param for `ascending` because we might as well do it now